### PR TITLE
Terraform - Converted Lambda functions and API Gateway for send email

### PR DESCRIPTION
**Pull request by thomaschin35**

### CODE REVIEW

File: dlp-terraform/lambda/source-files/lambda_function.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    except ClientError as e:
+        return(e.response["Error"]["Message"])
+    else:
+        return("Email sent! Message ID:"),
+        return(response["MessageId"])
+
+def lambda_handler(event, context):
+    print(event)
+    myDict = json.loads(event['body'])
+    print(event)
```
Comment (farisdurrani) [inline dlp-terraform/lambda/source-files/lambda_function.py:81]: redundant code

File: dlp-terraform/lambda/source-files/lambda_function.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            RawMessage={
+                "Data": msg.as_string(),
+            },
+        )
+    # Display an error if something goes wrong.
+    except ClientError as e:
+        return(e.response["Error"]["Message"])
+    else:
+        return("Email sent! Message ID:"),
+        return(response["MessageId"])
```
Comment (farisdurrani) [inline dlp-terraform/lambda/source-files/lambda_function.py:76]: Why two returns?

**Review by farisdurrani — CHANGES_REQUESTED** [review]

File: dlp-terraform/lambda/source-files/lambda_function.py (hunk 2 above)
Comment (karkir0003) [reply dlp-terraform/lambda/source-files/lambda_function.py:76]: @thomaschin35 why two returns?

File: dlp-terraform/lambda.tf [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  output_path = "outputs/send_email.zip"
+  source_dir  = "lambda/source-files"
+}
+
+resource "aws_lambda_function" "send_email" {
+  # If the file is not in the current working directory you will need to include a
+  # path.module in the filename.
+  filename      = "outputs/send_email.zip"
+  function_name = "send_email"
+  role          = "arn:aws:iam::521654603461:role/send_email"
```
Comment (karkir0003) [inline dlp-terraform/lambda.tf:13]: @thomaschin35 did u test the deploy and run the python unit tests to see if stuff works. you can also try training simple iris model to validate

**Review by karkir0003 — CHANGES_REQUESTED** [review]
address comments from faris and I

File: dlp-terraform/lambda/preprocess_lambda_function.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import json
+import pandas as pd
+import numpy as np
+import traceback
+
+def preprocess(data, code):
```
Comment (karkir0003) [inline dlp-terraform/lambda/preprocess_lambda_function.py:6]: i understand that you needed to expose the handler we wrote in the lambda in order to be terraform compatible. we should setup some special check that enforces certain people to review this handler if someone inadvertently changes this file. You can enforce such restriction using the `CODEOWNERS` file @thomaschin35

File: dlp-terraform/lambda/source-files/lambda_function.py (hunk 2 above)
Comment (karkir0003) [reply dlp-terraform/lambda/source-files/lambda_function.py:76]: @thomaschin35

File: dlp-terraform/lambda/source-files/lambda_function.py (hunk 1 above)
Comment (karkir0003) [reply dlp-terraform/lambda/source-files/lambda_function.py:81]: @thomaschin35 agree with faris on this one

File: dlp-terraform/lambda/source-files/lambda_function.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    # parent container.
+    msg.attach(msg_body)
+
+    try:
+        # Provide the contents of the email.
+        response = client.send_raw_email(
+            Source=SENDER,
+            Destinations=[email_address],
+            RawMessage={
+                "Data": msg.as_string(),
```
Comment (karkir0003) [inline dlp-terraform/lambda/source-files/lambda_function.py:68]: enforce codeowners restriction for this file as well. 

You can mark yourself (@thomaschin35 and @avayedawadi as CODEOWNERS for this file)

**Review by karkir0003 — CHANGES_REQUESTED** [review]
address comments

File: backend/common/email_notifier.py [hunk 6]
```diff
@@ -17,6 +17,7 @@ def send_email(email_address, subject="", body_text="", attachment_array=[]):
     regex = re.compile(
         r"([A-Za-z0-9]+[.-_])*[A-Za-z0-9]+@[A-Za-z0-9-]+(\.[A-Z|a-z]{2,})+"
     )
+    print("I'm before full match")
```
Comment (karkir0003) [inline backend/common/email_notifier.py:20]: remove

File: backend/common/email_notifier.py [hunk 7]
```diff
@@ -27,9 +28,9 @@ def send_email(email_address, subject="", body_text="", attachment_array=[]):
             my_string = my_string.decode("utf-8")
         base64Array.append(my_string)
 
-    url = "https://kwado68i00.execute-api.us-west-2.amazonaws.com/send_email"
+    url = "https://6amfyprxh9.execute-api.us-west-2.amazonaws.com/default/send_email"
     params = {"recipient": email_address, "subject": subject, "body_text": body_text}
     body = {"attachment_array": base64Array, "file_names": fileNames}
-
+    print("Before post")
```
Comment (karkir0003) [inline backend/common/email_notifier.py:34]: remove

**Review by karkir0003 — CHANGES_REQUESTED** [review]
address minor comments

**Comment by karkir0003** [comment]
@thomaschin35 make sure you update the wiki with description of how terraform works, why we use it (advantages), setup for developers who are touching terraform

**Review by karkir0003 — APPROVED** [review]

**Review by farisdurrani — APPROVED** [review]

**Comment by farisdurrani** [comment]
@thomaschin35 you may go ahead and merge this PR

**Pull request merged**